operator-framework/src/test/java/io/javaoperatorsdk/operator/sample/externalstate/ExternalStateReconciler.java

```diff
@@ -0,0 +1,125 @@
+import io.javaoperatorsdk.operator.processing.event.source.EventSourceStartPriority;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+import io.javaoperatorsdk.operator.processing.event.source.polling.PerResourcePollingEventSource;
+import io.javaoperatorsdk.operator.support.ExternalIDGenServiceMock;
+import io.javaoperatorsdk.operator.support.ExternalResource;
+import io.javaoperatorsdk.operator.support.TestExecutionInfoProvider;
+
+@ControllerConfiguration()
+public class ExternalStateReconciler
+    implements Reconciler<ExternalStateCustomResource>, Cleaner<ExternalStateCustomResource>,
+    EventSourceInitializer<ExternalStateCustomResource>, KubernetesClientAware,
+    TestExecutionInfoProvider {
+
+  public static final String ID_KEY = "id";
+  private final AtomicInteger numberOfExecutions = new AtomicInteger(0);
+
+  private ExternalIDGenServiceMock externalService = ExternalIDGenServiceMock.getInstance();
+  private KubernetesClient client;
+
+  InformerEventSource<ConfigMap, ExternalStateCustomResource> configMapEventSource;
+  PerResourcePollingEventSource<ExternalResource, ExternalStateCustomResource> externalResourceEventSource;
+
+  @Override
+  public UpdateControl<ExternalStateCustomResource> reconcile(
+      ExternalStateCustomResource resource, Context<ExternalStateCustomResource> context) {
+    numberOfExecutions.addAndGet(1);
+
+    var externalResource = context.getSecondaryResource(ExternalResource.class);
+    externalResource.ifPresentOrElse(r -> {
+      if (!r.getData().equals(resource.getSpec().getData())) {
+        updateExternalResource(resource, r);
+      }
+    }, () -> {
+      if (externalResource.isEmpty()) {
+        createExternalResource(resource);
+      }
+    });
+
+
+    return UpdateControl.noUpdate();
+  }
+
+  private void updateExternalResource(ExternalStateCustomResource resource,
+      ExternalResource externalResource) {
+    var newResource = new ExternalResource(externalResource.getId(), resource.getSpec().getData());
+    externalService.update(newResource);
+    externalResourceEventSource.handleRecentResourceUpdate(ResourceID.fromResource(resource),
+        newResource, externalResource);
+  }
+
+  private void createExternalResource(ExternalStateCustomResource resource) {
+    var createdResource =
+        externalService.create(new ExternalResource(resource.getSpec().getData()));
+    var configMap = new ConfigMapBuilder()
+        .withMetadata(new ObjectMetaBuilder()
+            .withName(resource.getMetadata().getName())
+            .withNamespace(resource.getMetadata().getNamespace())
+            .build())
+        .withData(Map.of(ID_KEY, createdResource.getId()))
+        .build();
+    configMap.addOwnerReference(resource);
+    client.configMaps().resource(configMap).create();
+
+    var primaryID = ResourceID.fromResource(resource);
+    configMapEventSource.handleRecentResourceCreate(primaryID, configMap);
```

Comment:
This method shouldn't be called directly, imo. It shouldn't even be part of the public interface.

Reply:
In this case it is needed, since we want to be strongly consistent regarding the event source. 
(For long term we might want to implement these features directly to the client, similarly as in go)

Reply:
Sure, but that's not up to the user to ensure that consistency. If the user has to enforce that, then we need to redesign things because the user shouldn't even be aware of that part as it's internal to how the SDK works.

Reply:
Dependent resources will hide this, but some might do it on lower level. The thing is that without that it just don't work, (In all cases). We need probably do some work on these levels in the future. But this is how it is now, hiding that layer would won't solve the problem. And dependent resources using that, as other abstractions might use that user implement itself, so definitelly would not hide it.